src/main/java/org/opensearch/knn/index/codec/KNN80Codec/KNN80DocValuesConsumer.java

```diff
@@ -110,61 +111,72 @@ public void addKNNBinaryField(FieldInfo field, DocValuesProducer valuesProducer,
         throws IOException {
         // Get values to be indexed
         BinaryDocValues values = valuesProducer.getBinary(field);
-        KNNCodecUtil.Pair pair = KNNCodecUtil.getFloats(values);
-        if (pair.vectors.length == 0 || pair.docs.length == 0) {
-            logger.info("Skipping engine index creation as there are no vectors or docs in the documents");
-            return;
-        }
-        long arraySize = calculateArraySize(pair.vectors, pair.serializationMode);
-        if (isMerge) {
-            KNNGraphValue.MERGE_CURRENT_OPERATIONS.increment();
-            KNNGraphValue.MERGE_CURRENT_DOCS.incrementBy(pair.docs.length);
-            KNNGraphValue.MERGE_CURRENT_SIZE_IN_BYTES.incrementBy(arraySize);
-        }
-        // Increment counter for number of graph index requests
-        KNNCounter.GRAPH_INDEX_REQUESTS.increment();
-        // Create library index either from model or from scratch
-        String engineFileName;
-        String indexPath;
-        NativeIndexCreator indexCreator;
-        final KNNEngine knnEngine = getKNNEngine(field);
-        if (field.attributes().containsKey(MODEL_ID)) {
+        KNNCodecUtil.Pair pair = null;
+        try {
+            pair = KNNCodecUtil.getFloats(values);
+            if (pair.getVectorAddress() == 0 || pair.docs.length == 0) {
+                logger.info("Skipping engine index creation as there are no vectors or docs in the segment");
```

Comment:
do we need to differentiate between those two conditions for if and log them individually with corresponding message?

Reply:
I think we can keep it like this only. not having docIds and still having vectors is not a use case for us.